app/services/submissions_sort_and_filter_service.rb

```diff
@@ -77,4 +78,11 @@ def apply_sorting
       @submissions = @submissions.order(id: :asc)
     end
   end
+
+  def apply_includes
+    @submissions = @submissions.public_send(
+      @params[:sort]&.include?('assignees') ? :preload : :includes,
+      evaluator_submission_assignments: [:evaluator, :evaluation]
+    )
```

Comment:
I would avoid `public_send` if it's not needed, it's better to be explicit (see suggestion).  Maybe the conditional isn't necessary and we can use `preload` for both cases though? 🤔 
```suggestion
    @submissions = if @params[:sort]&.include?('assignees')
      @submissions.preload(evaluator_submission_assignments: [:evaluator, :evaluation])
    else
      @submissions.includes(evaluator_submission_assignments: [:evaluator, :evaluation])
    end
```

Reply:
I saw a Bullet unoptimized query locally and tested this, it seems to fix the optimization for all sort/filter params
```suggestion
    @submissions = @submissions.preload(
      evaluator_submission_assignments: [:evaluator, {evaluation: :evaluation_scores} ]
    )
```